pkg/repository/cachedep.go

```diff
@@ -38,7 +38,7 @@ func newCacheDependencyManager(conn db.Connection, debug bool) *cacheDependencyM
 	}
 }
 
-func (cdm *cacheDependencyManager) transactional(desc string, dbOps func() error) error {
+func (cdm *cacheDependencyManager) transactional(desc string, dbOps func(tx *db.TxConnection) error) error {
 	if err := db.Transaction(cdm.conn, dbOps, cdm.logger); err != nil {
```

Comment:
Current implementation is non-sense: the `cachedep.conn` has to be removed. The `Exec()` has to get the DB-connection passed form outside - this ensures the correct TX will be used.

Reply:
Fixed